src/formatters.ts

```diff
@@ -27,16 +27,16 @@ async function reloadFormatters(sourceRoot: string, context: vscode.ExtensionCon
   const name = extensionConfiguration("formatting").provider;
   const props = formatters[name];
 
-  const { tool, error } = await props.check();
-  if (error) {
-    getOutputChannel().appendLine(`Failed to enable formatter ${name}: ${error}`);
+  const checkResult = await props.check();
+  if (checkHasError(checkResult)) {
```

Comment:
What do you think about making checkHasError a method on ToolCheckResult?

Reply:
That would work too, it would need to be a class, but it would look better 🤔

Reply:
That would be my preferred way of going about this if you could :smile:. Thanks for your work getting these 2 PRs over the hump!

Reply:
I implemented it as class now (in a new commit, I can squash it later), the only thing that is not that nice, is that some getter functions have runtime error checking, since it's not possible to narrow class members with the `xy is XY` constraint as before, but for the 2 places where it's used, this works fine and looks nicer